### PR TITLE
site: don't show progress bar for new templates

### LINKED ISSUE #5229: Creating a workspace from a new template shows NaN in build progress bar
Version: [`v0.12.9+2b6c229`](https://github.com/coder/coder/commit/2b6c229e4e2e94776500fe366b538afb6c24c199)

image

### PR DESCRIPTION
**Pull request by ammario**
Resolves #5229

### CODE REVIEW

File: site/src/components/WorkspaceBuildProgress/WorkspaceBuildProgress.tsx [hunk 1]
```diff
@@ -102,6 +102,11 @@ export const WorkspaceBuildProgress: FC<WorkspaceBuildProgressProps> = ({
     setTimeout(updateProgress, 5)
   }, [progressValue, job, transitionStats])
 
+  // HACK: the codersdk type generator doesn't support null values, but this
+  // can be null when the template is new.
+  if ((transitionStats.P50 as number | null) === null) {
```
Comment (BrunoQuaresma) [inline site/src/components/WorkspaceBuildProgress/WorkspaceBuildProgress.tsx:107]: What does` transitionStats.P50` means?

Comment (ammario) [reply site/src/components/WorkspaceBuildProgress/WorkspaceBuildProgress.tsx:107]: PX in general refers the Xth percentile. So P50 refers to the 50th percentile.

Comment (BrunoQuaresma) [reply site/src/components/WorkspaceBuildProgress/WorkspaceBuildProgress.tsx:107]: I see we have a test case for when `transitionStats` is `undefined` but it is not reflected in the component props:
```
export interface WorkspaceBuildProgressProps {
  workspace: Workspace
  transitionStats: TransitionStats
}
```
When using `WorkspaceBuildProgressProps` is it possible for `transitionStats` to be `undefined`? I see the API returns the `transitionStats` and its values can be `undefined` or `null`, but If yes, `transitionStats` can be `undefined`, we have to update the component props to reflect that and maybe fix it in some other places. If not, we can just remove/update the test case to reflect the new data structure.

Comment (ammario) [reply site/src/components/WorkspaceBuildProgress/WorkspaceBuildProgress.tsx:107]: It's not possible for transitionStats to be undefined. The backend always sends it. The values should be typed as `number | null` but I need to edit the type generator to make it so.

Comment (BrunoQuaresma) [reply site/src/components/WorkspaceBuildProgress/WorkspaceBuildProgress.tsx:107]: Fixed the storybook. I will merge this when the storybook tests pass

**Review by BrunoQuaresma — APPROVED** [review]
LGTM

**Pull request merged**